inlang/source-code/sdk/src/loadProject.ts

```diff
@@ -92,9 +101,13 @@ export async function loadProject(args: {
 			// }
 
 			loadSettings({ settingsFilePath: projectPath + "/settings.json", nodeishFs })
-				.then((settings) => setSettings(settings))
+				.then((settings) => {
+					setSettings(settings)
+					markSettingsAsLoaded()
+				})
 				.catch((err) => {
 					markInitAsFailed(err)
+					markSettingsAsFailed(err)
 				})
 		})
```

Comment:
@martin-lysk can you confirm that this effect is not really doing anything and could be removed, allowing the loadSettings to be awaited?

Reply:
good catch - this effect doesn't have a reference to a signal.
Context: i guess this one was meant to watch on an effect produced by a file watcher on the settings file in the TODO the line below.

Reply:
Shall we keep the effect and introduce settings file watching instead or introduce settings reactivity with the new sdk?

Reply:
no, let's not fix this now - it will adds more complexity to watch files and differentiate between file-sourced vs. api-sourced changes.

#### notes from our 1:1 discussion:
- the planned intent (as mentioned in the TODO) was to watch the fs for settings changes but this was never implemented.
- awaiting and immediately throwing an error on load if the settings are invalid would make the code simpler. @jldec prefers this approach because it would simplify the startup logic 
- we need to handle runtime changes (after initial load) for settings as well, including error cases.